SpatialGDK/Source/SpatialGDKEditor/Private/Utils/LaunchConfigurationEditor.cpp

```diff
@@ -113,48 +140,42 @@ UTransientUObjectEditor* UTransientUObjectEditor::LaunchTransientUObjectEditor(c
 				[
 					SNew(SHorizontalBox)
 					+ SHorizontalBox::Slot()
-				.AutoWidth()
-				.Padding(2.0)
-				[
-					SNew(SButton)
-					.Text(ButtonCaption)
-				.OnClicked(FOnClicked::CreateStatic(&ExecuteEditorCommand, ObjectInstance, Function))
-				]
+					.AutoWidth()
+					.Padding(2.0)
+					[
+						SNew(SButton)
+						.Text(ButtonCaption)
+						.OnClicked(FOnClicked::CreateStatic(&ExecuteEditorCommand, ObjectInstance, Function))
+					]
 				];
 		}
 	}
 
 	TSharedRef<SWindow> NewSlateWindow = SNew(SWindow)
-		.Title(FText::FromString(EditorName))
+		.Title(FText::FromString(TEXT("Launch Configuration Editor")))
+		.ClientSize(FVector2D(600, 400))
 		[
 			SNew(SBorder)
 			.BorderImage(FEditorStyle::GetBrush(TEXT("PropertyWindow.WindowBorder")))
 			[
 				VBoxBuilder
 			]
 		];
-	
-	if (!ParentWindow.IsValid() && FModuleManager::Get().IsModuleLoaded("MainFrame"))
+
+	if (!InParentWindow.IsValid() && FModuleManager::Get().IsModuleLoaded("MainFrame"))
 	{
 		// If the main frame exists parent the window to it
 		IMainFrameModule& MainFrame = FModuleManager::GetModuleChecked<IMainFrameModule>("MainFrame");
-		ParentWindow = MainFrame.GetParentWindow();
+		InParentWindow = MainFrame.GetParentWindow();
 	}
 
-	if (ParentWindow.IsValid())
+	if (InParentWindow.IsValid())
 	{
-		FSlateApplication::Get().AddWindowAsNativeChild(NewSlateWindow, ParentWindow.ToSharedRef());
+		FSlateApplication::Get().AddModalWindow(NewSlateWindow, InParentWindow.ToSharedRef());
 	}
 	else
 	{
-		FSlateApplication::Get().AddWindow(NewSlateWindow);
+		FSlateApplication::Get().AddModalWindow(NewSlateWindow, nullptr);
 	}
-
-	NewSlateWindow->RegisterActiveTimer(0.5, FWidgetActiveTimerDelegate::CreateLambda([NewSlateWindow](double, float)
```

Comment:
I acknowledge this was a hack to get the right size for the windows and circumvent Slate's layout logic, but why did it need to be removed ?

Reply:
It resizes the window to the desired size after opened 0.5s. If changed to a modal window, this code is blocked until closing the window.
Give the windows a initialize size is simple to fix that.